pkg/schema/annotations.go

```diff
@@ -170,26 +174,36 @@ func collectAnnotations(item yamlmeta.Node) ([]Annotation, error) {
 	return anns, nil
 }
 
-func processOptionalAnnotation(node yamlmeta.Node, optionalAnnotation template.AnnotationName) (Annotation, error) {
-	nodeAnnotations := template.NewAnnotations(node)
-
-	if nodeAnnotations.Has(optionalAnnotation) {
-		ann, _ := nodeAnnotations[optionalAnnotation]
+func collectValueAnnotations(item yamlmeta.Node, t yamlmeta.Type) ([]Annotation, error) {
+	var anns []Annotation
 
-		wrappedValueType, err := inferTypeFromValue(node.GetValues()[0], node.GetPosition())
+	for _, annotation := range []template.AnnotationName{AnnotationNullable, AnnotationDefault} {
+		ann, err := processOptionalAnnotation(item, annotation, t)
 		if err != nil {
 			return nil, err
 		}
+		if ann != nil {
+			anns = append(anns, ann)
+		}
+	}
+	return anns, nil
+}
+
+func processOptionalAnnotation(node yamlmeta.Node, optionalAnnotation template.AnnotationName, t yamlmeta.Type) (Annotation, error) {
```

Comment:
```suggestion
func processOptionalAnnotation(node yamlmeta.Node, optionalAnnotation template.AnnotationName, effectiveType yamlmeta.Type) (Annotation, error) {
```
💭  "length of a variable 'ought' to be in proportion to the distance it is used"; this function is close to a "screen-full"; what role is this `yamlmeta.Type` playing here (answering that reveals intention)?

This propagates up and down the call stack, here:
- `NewDocumentType()` / `NewMapItemType()` / `NewArrayItemType()`
- `getValue()`
- `collectValueAnnotations()`
- `NewDefaultAnnotation()` (In a conversation we were having, I think I had said this was the "inferred type" ... and I was wrong. 😬 )

the actual name is (obviously) implementor's choice:
- `effectiveType`
- `typeOfValue`
- `actualType`
- ...